packages/manager/modules/iam/src/components/createResourceGroup/createResourceGroup.controller.js

```diff
@@ -186,13 +185,30 @@ export default class CreateResourceGroupController {
    * @param {boolean} success
    */
   onDeleteEntityGoBack({ success }) {
-    if (success) {
-      this.model.resources = this.model.resources?.filter(
-        (resource) => !this.deletion.data.resources.includes(resource),
-      );
-    }
+    this.deletion.promise
+      .then(() =>
+        this.$timeout(() => {
+          const {
+            data: { resources: deletedResources },
+          } = this.deletion;
+
+          if (success) {
+            this.model.resources = {
+              ...this.model.resources,
+              selection:
+                this.model.resources.selection?.filter(
+                  ({ id }) =>
+                    !deletedResources.find((resource) => id === resource.id),
+                ) || [],
+            };
+          }
+        }),
+      )
+      .then(() => {
+        this.deletion = null;
```

Comment:
What about if `this.deletion` fails, `this.deletion` should also be set to null or not ?

Reply:
The promise can not fail with the following line
```javascript
this.deletion.resolve(success);
```
But I agree this should be set in a `finally` statement